### ISSUE TITLE
Facebook Auth is Working Well on Expo, but Not After Deploying and Testing the iOS app Through TestFlight

### ISSUE THREAD

**Issue by jyoo** [expo/expo#2989]
### Environment

Expo CLI 2.6.5 environment info:
    System:
      OS: macOS High Sierra 10.13.6
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 10.10.0 - /usr/local/bin/node
      Yarn: 1.9.4 - /usr/local/bin/yarn
      npm: 6.4.1 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Xcode: 10.1/10B61 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^30.0.1 => 30.0.1 
      react: 16.3.1 => 16.3.1 
      react-native: https://github.com/expo/react-native/archive/sdk-30.0.0.tar.gz => 0.55.4 
      react-navigation: ^2.16.0 => 2.16.0 
    npmGlobalPackages:
      expo-cli: 2.6.5

iOS

### Steps to Reproduce

1. I made an app which uses Expo Facebook Auth (using Expo.Facebook.logInWithReadPermissionsAsync)
2. Using expo build:ios, I have deployed the app so that I can test through TestFlight
3. When I tried to login via Facebook, I could not see any response.

### Expected Behavior

It should be possible for a user to log in or sign up via Facebook

### Actual Behavior

If I use Expo app to test my app before deploying (expo build:ios), I can sign in through Facebook. Once I press the Facebook login button, a page appears so that I can enter my facebook ID and password and I receive the token to use Firebase Auth. The problem is that when I deployed the app, downloaded it through TestFlight, and pressed the Facebook login button, nothing shows up.

Code: 

```js
const { type, token } = await Expo.Facebook.logInWithReadPermissionsAsync(someNumber, { permissions: ['public_profile', 'email']});

		if (type === 'success') {
			const credential = firebase.auth.FacebookAuthProvider.credential(token);
			firebase.auth().signInAndRetrieveDataWithCredential(credential)
			.then((user) => {
				// some codes
			}).catch((err) => {
				// somecodes
			})
		}
```
### Reproducible Demo

It does not make any problem on Expo.

**Comment by vcebotari** [comment]
same environment, same problem

**Comment by esamelson** [comment]
Hi, can someone provide a full repro to this (ideally a snack) that we can copy + paste and build a standalone app with? So far we've been unable to reproduce this issue, and having an easy repro case would be a big help in getting this resolved quickly. Thanks!

**Comment by freehunter** [comment]
I’m running into the same issue. Facebook auth works from Expo but when I generate an app and send it to Apple, the Test Flight version of the app doesn’t work. Oddly enough, it doesn’t work on my iPhone XS or my iPhone 8, but it does work properly on my iPhone 5s and iPhone SE with iOS 12.1.1.

I get right up to

```
let { type, token } = await Facebook.logInWithReadPermissionsAsync('[App ID Removed]', {
    permissions: ['public_profile', 'email']
})
```

And that’s where the code stops running on the 8 and XS. But again, runs just fine on an iPhone 5s and iPhone SE after downloading from Test Flight. Apps I've already built are still working correctly, but any new app I build that includes the above line of code fails on the XS and 8 but not the 5s or SE.

**Comment by Alisan26** [comment]
Same problem, facebooklogin works fine in simulator but I build ios and thrown away to testflight, facebooklogin wont work pressed the Facebook login button, nothing shows up.
I use expo sdk 30 , I tested my iphone 6 12.0.1 device
My code is below
```
const doFacebookLogin = async dispatch => {
    let { type, token } = await Facebook.logInWithReadPermissionsAsync(
      'App Id',
      {
        permissions: ['public_profile']
      }
    );
    if (type === 'cancel') {
      return dispatch({ type: FACEBOOK_LOGIN_FAIL });
    }
  
    await AsyncStorage.setItem('fb_token', token);
    dispatch({ type: FACEBOOK_LOGIN_SUCCESS, payload: token });
};
```

**Comment by thomaskjellberg** [comment]
> Hi, can someone provide a full repro to this (ideally a snack) that we can copy + paste and build a standalone app with? So far we've been unable to reproduce this issue, and having an easy repro case would be a big help in getting this resolved quickly. Thanks!

You can test this @esamelson , having the same issue. Working fine until Testflight, where nothing happens when clicking the button.

https://snack.expo.io/BkFJ9ibZE

Also added:  "facebookAppId", "facebookScheme"  & "facebookDisplayName" in app.json.

**Comment by harrymelka** [comment]
Same for me

**Comment by harrymelka** [comment]
Same for me,

I tried with expo v30 and v31 and it didn't fix the issue

**Comment by fsocietyomer** [comment]
Same. 
I was also using logInWithReadPermissionsAsync and nothing happens on iOS when i press the login button.

**Comment by sjchmiela** [comment]
Hm… 🤔 I've just created a Turtle-built app on SDK32 with `App.js` code similar to https://snack.expo.io/BkFJ9ibZE. My `app.json` looks like:
```jsonc
{
  "expo": {
    "name": "sdk32-facebook-login-ios",
    "slug": "sdk32-facebook-login-ios",
    "privacy": "public",
    "sdkVersion": "32.0.0",
    "platforms":  [ /* ... */ ],
    "version": "1.0.0",
    "orientation": "portrait",
    "icon": "./assets/icon.png",
    "splash": { /* ... */ },
    "updates": { /* ... */ },
    "assetBundlePatterns": { /* ... */ },
    "facebookAppId": "629712900716487",
    "facebookDisplayName": "expotest",
    "facebookScheme": "fb629712900716487",
    "android": {
      "package": "test.sjchmiela.sdk32facebooklogin"
    },
    "ios": {
      "buildNumber": "2",
      "supportsTablet": true,
      "bundleIdentifier": "test.sjchmiela.sdk32facebooklogin",
      "infoPlist": {
        "LSApplicationQueriesSchemes": ["fbapi", "fb-messenger-share-api", "fbauth2", "fbshareextension"]
      }
    }
  }
}
```

…and logging in succeeds, alerting proper username. 😱 I think adding the `LSApplicationQueriesSchemes` may be pretty non-standard, so I'll check again without this property and see if I'm able to reproduce this issue.

**Comment by harrymelka** [comment]
Just updated the expo version to v32 and it's working on dev mode on expo like before but not on publish mode on the real app.
I added everything `LSApplicationQueriesSchemes` with the full array and nothing worked.
It goes directly on the catch in the facebook login.

**Comment by sjchmiela** [comment]
Thanks, @harrymelka!

1. What does the caught error contain?
2. What device with what iOS version do you use to test the newly built application?

**Comment by harrymelka** [comment]
```javascript
async loginWithFacebook(): Promise<void> {
        try {
            // Real
            const { type, token } = await Facebook.logInWithReadPermissionsAsync('TOKEN', { permissions: ["public_profile", "email"], behavior: "web" }
            );
            switch (type) {
                case "success": {
                    // Get the user's name using Facebook's Graph API
                    const response = await fetch(`https://graph.facebook.com/me?access_token=${token}&fields=email,id,name,picture.type(large),gender,friends`);
                    const profile = await response.json();
                    const credential = firebase.auth.FacebookAuthProvider.credential(token);
                    // Sign in with credential from the Facebook user.
                    firebase.auth().signInWithCredential(credential).then((result) => {
                          .........
                        }).catch((error) => {
                            console.log(error);
                        })
                    }).catch((error) => {
                        // Handle Errors here.
                        console.log(error);
                    });

                    break;
                }
                case "cancel": {
                    Alert.alert(
                        "Annulé!",
                        "Connexion Annulée",
                    );
                    break;
                }
                default: {
                    Alert.alert(
                        "Oops!",
                        "Erreur Connexion! 1",
                    );
                }
            }
        } catch (e) {
            Alert.alert(
                "Oops!",
                "La connexion a échoué avec Facebook! Nous allons l'examiner. Merci de créer un compte.",
            );
            console.log(e.message);
        }
    }
```
I tried it on iPhone 8, 8+, X, XsMax with ios 12 mostly previous and newest versions.

#app.json

```json
{
  "expo": {
    ....
    "sdkVersion": "32.0.0",
    "platforms": [
      "ios",
      "android"
    ],
    "facebookScheme": "fb.....",
    "facebookAppId": ".....",
    "ios": {
      "bundleIdentifier": "...",
      "infoPlist": {
        "NSCameraUsageDescription": "L'application a besoin de la permission pour pouvoir scanner les codes barre.",
        "NSLocationWhenInUseUsageDescription": "L'application a besoin de la permission pour pouvoir vous aider à trouver les commerces proche de vous.",
        "LSApplicationQueriesSchemes": [
          "waze",
          "fbapi",
          "fb-messenger-share-api",
          "fbauth2",
          "fbshareextension"
        ]
      }
    }
  }
}
```
#package.json

```json
{
  "name": "my-project",
  "main": "node_modules/expo/AppEntry.js",
  "private": true,
  "scripts": {
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "eject": "expo eject",
    "test": "node ./node_modules/jest/bin/jest.js --watchAll"
  },
  "jest": {
    "preset": "jest-expo"
  },
  "dependencies": {
    "@expo/samples": "2.1.1",
    "@expo/vector-icons": "^8.0.0",
    "expo": "^32.0.0",
    ....
    "react": "16.5.0",
    "react-native": "https://github.com/expo/react-native/archive/sdk-32.0.0.tar.gz",
    .....
    "yarn": "^1.12.3"
  },
  "devDependencies": {
    "jest-expo": "32.0.0"
  }
}
```
Here is what I have on the app, the versions and configs

If you have any ideas, I tried a lot of stuff. My facebook configs work perfectly on another app that use expo but the version is v29 so it's really in facebook expo login.

Thank you

**Comment by andreiStoicescu** [comment]
I am having the same problem on expo 30. It's not working on iPhone 8 but it is working on iPhone SE.
Did anyone managed to get it working without upgrading? Or checked if the login works on testflight with AuthSession instead of Facebook.logInWithReadPermissionsAsync ?

**Comment by hughbe** [comment]
Did anyone manage to fix this? We really need this feature for our app to work!

**Comment by hughbe** [comment]
Can't upgrade to v31 sadly due to https://github.com/expo/expo/issues/3189
Trying out setting the behaviour to 'web' on expo!

**Comment by fsocietyomer** [comment]
I've upgraded the expo SDK to 32 according to https://docs.expo.io/versions/latest/workflow/upgrading-expo-sdk-walkthrough and it solved the problem!!!! thanks @sjchmiela !

**Comment by sjchmiela** [comment]
This sounds a little bit too good to be true… 😅 (Although… we did upgrade FBSDK in Expo SDK 32 to 4.38.1… 🤔).

It looks like `react-native-facebook-login` may suffer from similar problem — https://github.com/magus/react-native-facebook-login/issues/330.

Has anyone else been able to fix the issue by upgrading too?

**Comment by ncailleux** [comment]
Same issue with v30

And i got some issues when upgrading to v31 or v32 for now (v32 full white screen and nothing happend except http request spamming, and v31 react-native-maps-directions issue, nice one too)
So couldn't try this "fix"

Anyone for another solution in v30 for now?

**Comment by andreiStoicescu** [comment]
i managed to make it work in expo v30 with AuthSession
` let result = await AuthSession.startAsync({
    authUrl: facebookAuthUrl + '&redirect_uri=' + redirectUrl + '&response_type=token'
  })
`

**Comment by hughbe** [comment]
What is `facebookAuthUrl`

**Comment by andreiStoicescu** [comment]
@hughbe you can get the authUrl from facebook documentation

**Comment by sjchmiela** [comment]
Interesting… 🤔 Is there anyone for whom upgrading to SDK32 didn't help solve the issue?

**Comment by IAmdeGroot** [comment]
@sjchmiela Im on SDK32 and its not working on testflight. In simulator I get an alert saying "Expo wants to use Facebook to sign in" but its never showing on testflight so I can't proceed.

**Comment by harrymelka** [comment]
So I updated again to V32 and it fix the issue but I had to send the app again to apple and not publish with expo. I had to uninstall the app and reinstall it on my phone for it to work.

Hope it helps

**Comment by snollygolly** [comment]
@sjchmiela I'm on SDK 32, and I'm using `AuthSession` from expo to login with Auth0.  Same issue as everyone else is describing.  Everything works fine while testing on the simulator and actual device.  Just deployed to TestFlight and nothing happens when I try to log in now.  This is my first time using expo, so I'm not entirely sure how to debug from here, but I'm happy to provide whatever you need.

**Comment by jakeprins** [comment]
I upgraded from v31 to v32 and still have this problem. Only after deploying and testing on TestFlight it's not working. Really need this to be fixed! Has anyone got other suggestions?

**Comment by daybreaker** [comment]
Can anyone not on SDK 32 say if the previous app.json solution worked?

We can't upgrade to SDK 32 because it broke some stuff with postMessage inside WebViews.

**Comment by nickjuntilla** [comment]
I got this working after adding what GBGituser did plus switching login behavior from 'web' to default.

**Comment by cmaronchick** [comment]
@nickjuntilla @GBGitUser Do you mind sharing what's different, or perhaps, what I need to add?

Thanks!
```

{
  "expo": {
    "name": "APP NAME",
    "description": "APP DESCRIPTION",
    "slug": "APP SLUP",
    "privacy": "public",
    "sdkVersion": "32.0.0",
    "ios": {
      "bundleIdentifier": "com.IDENTIFIER",
      "supportsTablet": true
    },
    "android": {
      "package": "com.IDENTIFIER"
    },
    "platforms": [
      "ios",
      "android"
    ],
    "version": "0.0.2",
    "orientation": "portrait",
    "icon": "./assets/images/icon.png",
    "splash": {
      "image": "./assets/images/splash.png",
      "resizeMode": "contain",
      "backgroundColor": "#ffffff"
    },
    "updates": {
      "fallbackToCacheTimeout": 0
    },
    "assetBundlePatterns": [
      "**/*"
    ]
  }
}
```

**Comment by bnft** [comment]
@cmaronchick Switch 

`Expo.Facebook.logInWithReadPermissionsAsync(
      social.fb_app_id,
      {
        permissions: ['public_profile', 'email'],
        behavior: 'web'
      }
    );`

to: 

`Expo.Facebook.logInWithReadPermissionsAsync(
      social.fb_app_id,
      {
        permissions: ['public_profile', 'email'],
      }
    );`

That's what I had to do. That triggers the default facebook login behavior. After that you will probably need to add 

`"facebookScheme": "...",
    "facebookAppId": "...",
    "facebookDisplayName": "...",`

Properties to your app.json file and you may need to submit new builds to the app store as well.

**Comment by nickjuntilla** [comment]
@cmaronchick Switch 

`Expo.Facebook.logInWithReadPermissionsAsync(
      social.fb_app_id,
      {
        permissions: ['public_profile', 'email'],
        behavior: 'web'
      }
    );`

to: 

`Expo.Facebook.logInWithReadPermissionsAsync(
      social.fb_app_id,
      {
        permissions: ['public_profile', 'email']
      }
    );`

That's what I had to do. That triggers the default facebook login behavior. After that you will probably need to add 

`"facebookScheme": "...",
    "facebookAppId": "...",
    "facebookDisplayName": "...",`

Properties to your app.json file and you may need to submit new builds to the app store as well.

**Comment by cmaronchick** [comment]
Thanks! It just occurred to me that I'm using AuthSession rather than Expo.Facebook, so I don't think that we're talking about the same thing, right?

**Comment by JorgeMoralesLopez** [comment]
I have the same problem, I just need to get the facebookScheme, I downloaded the firefight info.plist file and put it in the root of the project. How can I get facebookScheme, through xCode, is provided by firebase ??
I'm a bit involved with these configurations...

**Comment by cmaronchick** [comment]
Ok, I have this fixed. Your Facebook scheme is fb preceding your Facebook App ID.

The fix is to add 'scheme': 'com.DOMAIN' to the app.json file, and that should do it.

**Comment by JorgeMoralesLopez** [comment]
Simple as that?

**Comment by cmaronchick** [comment]
That's what got it working for me. YMMV.

**Comment by ctur** [comment]
I have upgraded my expo and It's **not** working on expo version 32.0.0 and 32.0.2

**Comment by cmaronchick** [comment]
@ctur Can you share any more information?

**Comment by ctur** [comment]
@cmaronchick Sure, 
Using default 'system' for login to fb.
`Expo.Facebook.logInWithReadPermissionsAsync( social.fb_app_id, { permissions: ['public_profile', 'email'] } );` 

and my app.json is exact same with @GBGitUser. Facebook scheme is Facebook ID preceding 'fb'. Upgraded expo, react native and react exactly as documentation mentioned.

@cmaronchick i don't really understand your fix 'scheme':com.DOMAIN, what is DOMAIN?
Also i am not using Firebase or something else. FB Login working both on DEV and Production also on Play Store version as expected for Android. But not working on Testflight.

**Comment by cmaronchick** [comment]
Hi @ctur, Ok, I wasn't able to utilize Expo.Facebook, so I can't advise you on that.

Regarding DOMAIN, I used the same value as bundleIdentifier for Android and iOS in the Scheme value. Hope that that's helpful.

**Comment by ctur** [comment]
Hi, @cmaronchick,and  thanks for answering and trying to help. DOMAIN as bundleidentifier makes sense i'll try that.

**Comment by ctur** [comment]
For me the problem was related to login behavior, i was using 'system' (default) behavior for both Android and IOS but it wasn't working on IOS. I have changed to 'web' behavior and it's working now.

Also i have added this to infoPlist - app.json (not sure if it's correct way or made it work)
`"NSAppTransportSecurity": {
          "NSExceptionDomains": {
            "facebook.com": {
              "NSIncludesSubdomains": true,
              "NSExceptionRequiresForwardSecrecy": false
            },
            "fbcdn.net": {
              "NSIncludesSubdomains": true,
              "NSExceptionRequiresForwardSecrecy": false
            }
          }
        }`

**Comment by cmaronchick** [comment]
Congrats @ctur!

**Comment by ctur** [comment]
@sjchmiela Could you help sir?

**Comment by ctur** [comment]
I have fixed my problem for 'system' - default behavior. It was related to my navigatior styling. Facebook Login screen was openning but it was appearing behind the app. I can now confirm that everything working properly with Expo version 32 with @GBGitUser app.json settings.

**Comment by ekinalcar** [comment]
Hi everyone,

I've read all the answers above and tried to do some fixing myself. I've found out that when I change the default behavior, everything works fine. Is this the 'right' way to do so?

`const {type,token} = await Expo.Facebook.logInWithReadPermissionsAsync('<APP_ID>', {permissions: ['public_profile','email']});`
to
`const {type,token} = await Expo.Facebook.logInWithReadPermissionsAsync('<APP_ID>', {permissions: ['public_profile','email'], behavior: "web"});`

Thank you.

**Comment by ctur** [comment]
Hello @ekinalcar, not sure what do you mean about right but i would say that it sounds good for me, https://docs.expo.io/versions/latest/sdk/facebook/, as mentioned on docs, it 'Attempts to log in through the Facebook account currently signed in through the device Settings', on iOS 11+ fallback to native behavior which attempts to login through native Facebook App, otherwise fallback to browser which opens browser to login.

**Comment by ekinalcar** [comment]
hi @ctur thank you for your quick reply. the problem is when I set the behavior to 'native' after the facebook login , the screen won't switch back to my app. here is the process described with screenshots. only a blank page shows at the end of the process. 

Hope I made myself clear :/

![screen_1](https://user-images.githubusercontent.com/31273861/58560690-4e94ec80-822e-11e9-85da-38eba68c309e.PNG)
![screen_2](https://user-images.githubusercontent.com/31273861/58560771-78e6aa00-822e-11e9-973a-62459cfb3703.PNG)
![screen_3](https://user-images.githubusercontent.com/31273861/58560692-4f2d8300-822e-11e9-9b60-da3eac4456ae.PNG)
![screen_4](https://user-images.githubusercontent.com/31273861/58560693-4f2d8300-822e-11e9-93c6-4b4d738e3c1e.PNG)

**Comment by ctur** [comment]
Did you add `host.exp.Exponent` as mentioned here https://docs.expo.io/versions/latest/sdk/facebook/ to your Ios Bundle Id inside Apps page on Facebook developer portal? and also you should add your Bundle Id of the to the app.json inside project

**Comment by ekinalcar** [comment]
@ctur 

yes I've added the iOS bundle id as mentioned in the dev portal and in the app.json file

**Comment by ctur** [comment]
Could you control `facebookScheme` and maybe share app.json settings? @ekinalcar

**Comment by ekinalcar** [comment]
Sure. my app.json looks like this.

{
  "expo": {
    "name": "ChallengeApp",
    "slug": "challenge-app",
    "privacy": "public",
    "sdkVersion": "32.0.0",
    "platforms": [
      "ios",
      "android"
    ],
    "version": "1.0.0",
    "orientation": "portrait",
    "icon": "./assets/images/icon.png",
    "splash": {
      "image": "./assets/images/splash.png",
      "resizeMode": "cover",
      "backgroundColor": "#ffffff"
    },
    "updates": {
      "fallbackToCacheTimeout": 0
    },
    "assetBundlePatterns": [
      "**/*"
    ],
    "facebookScheme": "fb...",
    "facebookAppId": "...",
    "facebookDisplayName": "ChallengeApp",
    "ios": {
      "supportsTablet": true,
      "bundleIdentifier": "com.ChallengeApp",
      "icon": "./assets/images/icon.png",
      "infoPlist": {
        "NSCameraUsageDescription": "blabla.",
        "NSLocationWhenInUseUsageDescription": "blabla.",
        "LSApplicationQueriesSchemes": [
          "fbapi",
          "fb-messenger-share-api",
          "fbauth2",
          "fbshareextension"
        ]
      }
    },
    "description": "blabla"
  }
}

**Comment by ctur** [comment]
@ekinalcar everything looks as expected i guess, if Scheme AppID are correct, it might be better to check errors on,
`const {type,token} = await Expo.Facebook.logInWithReadPermissionsAsync('<APP_ID>', {permissions: ['public_profile','email']});`
control type, token maybe?

**Comment by ekinalcar** [comment]
I will check it as soon as possible and let you know. by the way my app works on expo simulator but not when I publish.

**Comment by ekinalcar** [comment]
@ctur I still couldn't find the problem. I went with the behavior: 'web' and it works perfectly on my phone.thanks anyways.

**Comment by SkyRune** [comment]
Hey I'am in **sdk 32**

And the login works perfectly on Android, but I keep having the same error message on IOS

**"Try to perform Facebook login with behavior 'native', but no facebook app id was provided. Specify Facebook app id in app.json or switch to "web" behavior **

It's seems that on IOS the login is not reading the appId from the logInWithReadPermissionsAsync() params ... So I guess it s reading it from app.json, but I specified it...

I tried with native,system, browser

Web works on IOS but I suppose it does not require the facebookappid

The IOS i'am testing on has a **version >11**
{
  "expo": {
    "name": "MyApp",
    "icon": "./src/assets/images/logo-app.png",
    "version": "1.0.29",
    "slug": "my-app-slug",
    "description": "",
    "privacy": "public",
    "sdkVersion": "32.0.0",
    "orientation": "portrait",
    "primaryColor": "#fff",
    "facebookScheme": "fb318850582360166",
    "facebookAppId": "318850582360166",
    "facebookDisplayName": "Peps",
    "packagerOpts": {
      "assetExts": [
        "ttf",
        "mp4"
      ],
      "sourceExts": [
        "ts",
        "tsx"
      ],
      "transformer": "node_modules/react-native-typescript-transformer/index.js"
    },
    "ios": {
      "bundleIdentifier": "com.skyrune.myapp",
      "appStoreUrl": "https://itunes.apple.com/app/peps-un-bar-une-promo/id1463105178",
      "infoPlist": {
        "UIBackgroundModes": [
          "location"
        ],
        "NSLocationWhenInUseUsageDescription": "We are asking for your location for blabla",
        "LSApplicationQueriesSchemes": [
          "fbapi",
          "fb-messenger-share-api",
          "fbauth2",
          "fbshareextension"
        ]
      }
    },
    "android": {
      "package": "com.skyrune.myapp",
      "versionCode": 29,
      "permissions": [
        "LOCATION",
        "ACCESS_COARSE_LOCATION",
        "ACCESS_FINE_LOCATION",
        "READ_INTERNAL_STORAGE",
        "WRITE_INTERNAL_STORAGE",
        "READ_EXTERNAL_STORAGE",
        "WRITE_EXTERNAL_STORAGE"
      ],
      "playStoreUrl": "https://play.google.com/store/apps/details?id=com.skyrune.myapp"
    },
    "splash": {
      "image": "./src/assets/images/splash.png",
      "backgroundColor": "#262637",
      "resizeMode": "contain"
    },
    "platforms": [
      "android",
      "ios"
    ]
  }
}

**Comment by musemind** [comment]
We also have the problem on iOS with the `native` behavior: FB-Login-Screen is white after Login. The app is not redirecting to our App. The User is stuck.

Changing the behavior to `web` fixes the issue, but users with installed Facebook App are not logged in automatically anymore... On Android. everything's fine.

**Comment by sharma-sanjeev156** [comment]
@musemind I am facing the same issue. did you find any workaround working for the `native` behavior?

**Comment by dfagneraythink** [comment]
@musemind @sharma-sanjeev156 Anybody fixed that?

**Comment by hinting** [comment]
I had the screen getting stuck issue (with behavior: native) as well at first for Expo client.  I realized the documentation for specifying "Bundle ID" is incorrect (host.exp.Exponent).  It should be host.exp.exponent (the same as the one used in Google) and it works.

**Comment by safciplak** [comment]
> Ok, I have this fixed. Your Facebook scheme is fb preceding your Facebook App ID.
> 
> The fix is to add 'scheme': 'com.DOMAIN' to the app.json file, and that should do it.

@cmaronchick ty so much

**Comment by dhavaljardosh** [comment]
I'm stuck as well.

Expo 33

For Facebook: 
`Facebook Login Error: Tried to perform Facebook login with behavior 'null', but no Facebook app id was provided. Specify app id in Info.plist or switch to 'web' behavior`

For Google:
`I'm not able to get any error yet`

**Comment by BC01719** [comment]
Here's what ended up working for me on a release channel for iOS:
![ezgif-6-3680e2607244](https://user-images.githubusercontent.com/29097404/67108090-be831f80-f19b-11e9-9902-59866b9b7672.gif)

DISCLAIMER: I have not tried android. 

What I did was match the info.plist of section 4. Configure Your Project of the [Facebook Login iOS](https://developers.facebook.com/docs/facebook-login/ios) to the infoPlist section in app.json.

I assume it does not work for me on android because I figure I would have to do something similar to the app.json. I have no access to a physical android phone to test the published version, so I do not know. I theorize that for android, you would have to alter the "android" configuration of app.json to align with the 4. Edit Your Resources and Manifest section [Facebook Login Android](https://developers.facebook.com/docs/facebook-login/android)

My setup:
iPhone 6S running iOS 13.1.2
Client Version 2.13.0.107059
Expo SDK 35

app.json:
```
{
  "expo": {
    "name": "XXX",
    "slug": "XXX",
    "privacy": "unlisted",
    "sdkVersion": "35.0.0",
    "platforms": [
      "ios",
      "android"
    ],
    "version": "1.0.0",
    "orientation": "portrait",
    "icon": "./assets/icon.png",
    "splash": {
      "image": "./assets/splash.png",
      "resizeMode": "contain",
      "backgroundColor": "#000000"
    },
    "updates": {
      "fallbackToCacheTimeout": 0
    },
    "assetBundlePatterns": [
      "**/*"
    ],
    "ios": {
      "supportsTablet": true,
      "infoPlist": {
        "CFBundleURLTypes": {
          "CFBundleURLSchemes": [
            "fb#######"
          ]
        },
        "FacebookAppId": "#######",
        "FacebookDisplayName": "XXX",
        "LSApplicationQueriesSchemes": [
          "fbapi",
          "fb-messenger-share-api",
          "fbauth2",
          "fbshareextension"
        ],
      },
    }
  }
}
```

Expo Facebook logInWithReadPermissionsAsync():
```
await Facebook.logInWithReadPermissionsAsync(FACEBOOK_APP_ID, { permissions: ['public_profile'] }).then(async (response) => {
    if (response.type === 'success') {
         // Success
    }
}.catch( ... )
```

**Comment by esamelson** [comment]
Hey all -- I believe this issue has been resolved, and you should be able to get Facebook auth working on supported SDK versions (currently, 33-35) by following all of the required configuration steps in the docs & filling out all the fields in app.json. I'm going to go ahead and close this. Please create a new issue with a full repro case if this still happens in any of the currently supported SDK versions. Thanks!
